lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -206,30 +213,29 @@ def cert_action(module, iam, name, cpath, new_name, new_path, state,
                                  original_path=opath, new_path=new_path, cert_body=ocert,
                                  upload_date=upload_date, expiration_date=exp, arn=arn)
             else:
-                changed=False
+                changed = False
                 module.exit_json(changed=changed, name=name, cert_path=opath, cert_body=ocert,
                                  upload_date=upload_date, expiration_date=exp, arn=arn,
                                  msg='No new path or name specified. No changes made')
         else:
-            changed=True
-            iam.upload_server_cert(name, cert, key, cert_chain=chain, path=cpath)
+            changed = True
+            iam.upload_server_cert(name, cert, key, cert_chain=cert_chain, path=cpath)
             opath, ocert, ocert_id, upload_date, exp, arn = cert_meta(iam, name)
             module.exit_json(changed=changed, name=name, cert_path=opath, cert_body=ocert,
                                  upload_date=upload_date, expiration_date=exp, arn=arn)
     elif state == 'absent':
         if name in orig_cert_names:
-            changed=True
+            changed = True
             iam.delete_server_cert(name)
             module.exit_json(changed=changed, deleted_cert=name)
         else:
-            changed=False
+            changed = False
             module.exit_json(changed=changed, msg='Certificate with the name %s already absent' % name)
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        state=dict(
-            default=None, required=True, choices=['present', 'absent']),
+        state=dict(default=None, required=True, choices=['present', 'absent']),
```

Comment:
I'm assuming type='str' is possibly also a default?

Is it preferred to have at least something defined in the parameter dict?  Or just leave it empty if there is nothing required?